flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/factories/TableFactoryUtil.java

```diff
@@ -83,4 +86,19 @@ public static ExternalCatalog findAndCreateExternalCatalog(Descriptor descriptor
 
 		return tableSink;
 	}
+
+	/**
+	 * Returns a table sink matching the descriptor.
+	 */
+	public static <T> TableSink<T> findAndCreateTableSink(CatalogTable table) {
```

Comment:
I actually wonder if we can just call `findAndCreateTableSink(table.toProperties()) ` in `HiveTableFactory` rather than adding the two new utils? `TableFactoryUtil` just need a set of properties, and doesn't seem to need to understand how a CatalogTable serialize itself into a set of properties

Reply:
Well, there is just one option. In that case, we need to make the other two util methods public, which doesn't see good. I'd think the way here is cleaner.